pcap/pcap.go

```diff
@@ -336,8 +379,27 @@ func (p *Handle) getNextBufPtrLocked(ci *gopacket.CaptureInfo) error {
 	}
 	var result NextError
 	for {
+		// test if we need to stop
```

Comment:
Let's do the call to pcap_next_ex FIRST in the loop, then fall back to the timeout only if we don't have a packet.

```go
for atomic.LoadUint32(&p.stop) == 0 {
  result = NextError(pcap_next_ex(...))
  if (we should error) {
    return err
  } else if (we have packet) {
    return packet
  }
  pcap_wait(...)
}
```

This way, we don't call an extraneous (expensive) `select` syscall in between every packet, only if there isn't one already on the queue.